src/docs/arch-rts.dox

```diff
@@ -1,113 +1,76 @@
-1. Rolling back unstable fast truncate
-2. Rolling back unstable updates
-
-All internal pages are traversed to rollback unstable fast truncate operations and
-leaf pages are traversed to remove the unstable updates.
+For each selected table, its pages are read into the cache. The unstable in-memory updates and the
+unstable historical versions from the history store are removed. The on-disk version is replaced by
+the latest stable in-memory update if any, otherwise by the latest stable version from the history
+store. If there is no stable in-memory updates and no stable version in the history store, the data
+store entry is completely removed. In the situation where the table has no timestamped updates, all
+the historical versions related to this table are removed from the history store as they are no
+longer required and the on-disk value is preserved.
 
-Once the leaf page is identified to rollback the updates, it is performed in the
-following order.
-1. Check smallest insert lists on the page
-2. Traverse through all the on-disk keys
-    a. Check update list
-    b. Check insert list
-    c. Check the on-disk version if no stable update found in the update list.
-3. Traverse through the reconciled pages to abort any history store updates.
+Finally, RTS is triggered on startup, shutdown or when the application is initiated. It is worth
+noting that it cannot run if there is any transaction activity happening in WiredTiger.
 
-@section rts-abort-update How rollback to stable aborts in-memory updates
+@section rts-pages Internal and leaf pages
 
-Traverse through all the updates in the update list and abort them until a stable
-update is found.
+On internal pages, RTS can roll back unstable fast truncate operations.
+On leaf pages, RTS can roll back unstable updates with the following steps:
 
-@section rts-abort-on-disk-update How rollback to stable aborts on-disk update
+1. Checks the smallest insert lists on the page
+2. Traverses through all the on-disk keys
+    a. Checks the update list
+    b. Checks the insert list
+    c. Checks the on-disk version if no stable update is found in the update list.
+3. Traverses through the reconciled pages to abort any history store updates.
 
-If the start time pair is not stable try to find a valid update from the history
-store that is stable to replace the on-disk version otherwise remove the on-disk
-key. In case if the stop time pair if exists and if its not stable, restore the
-on-disk update again into the update list.
+@section rts-abort-update How RTS aborts in-memory updates
 
-To remove any existing update, rollback to stable adds globally visible tombstone to
-the key update list and this key will get removed later during the reconciliation.
+RTS traverses through all the updates defined in the update list and aborts every update until a
+stable is found. To remove any unstable update, RTS adds globally a visible tombstone to the key
```

Comment:
To removal any unstable update `on-disk`. Adding an on-disk makes the sentence clear.

Reply:
The previous content was saying the following in the `How rollback to stable aborts on-disk update` section:

```
To remove any existing update, rollback to stable adds globally visible tombstone to the key update list and this key will get removed later during the reconciliation.
```

Your comment is in the `How RTS aborts in-memory updates` section. I may have mixed up the content between those two sections (`in-memory` vs `on-disk`). Should I really mention `on-disk` here?

Reply:
Yes, earlier it was talking about an on-disk version separately, so no need to mention the on-disk. Instead of mentioning an unstable update, how about like "To remove any key from the table, RTS adds globally ..."

Reply:
Thanks fixed in 013adf66d271d5c71e4585a025b894cbb5ef32ea. However, we mention `update list` in the same sentence. Is that correct or do we need to mention the insert list too? Since we are adding a tombstone to an existing key, I believe we are just talking about the update list?